### PR TITLE
feat: RSTUF Worker using MySQL as Database

### PR DESCRIPTION
**Pull request by kairoaraujo**
Adds support to MySQL Server as Database Server.

It include small adds to the tests and including a test set in FT using MySQL in the Worker

### CODE REVIEW

**Comment by kairoaraujo** [comment]
FYI @jkjell

File: docker-compose-mysql.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - RSTUF_BROKER_SERVER=redis://redis
+      - RSTUF_REDIS_SERVER=redis://redis
+      - RSTUF_SQL_SERVER=mysql+pymysql://root:secret@mysql:3306/rstuf
+    volumes:
+      - ./:/opt/repository-service-tuf-worker:z
+      - repository-service-tuf-storage:/var/opt/repository-service-tuf/storage
+      - ./tests/files/key_storage/:/var/opt/repository-service-tuf/key_storage
+    depends_on:
+      redis:
+        condition: service_healthy
```
Comment (MVrachev) [inline docker-compose-mysql.yml:71]: Do we want to check for `mysql` healthy state?
We do that for `postgress` in the standard `docker-compose.yml`.

File: docker-compose-mysql.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      context: .
+    entrypoint: "bash entrypoint-dev.sh"
+    environment:
+      - DATA_DIR=./data
+      - RSTUF_STORAGE_BACKEND=LocalStorage
+      - RSTUF_LOCAL_STORAGE_BACKEND_PATH=/var/opt/repository-service-tuf/storage
+      - RSTUF_ONLINE_KEY_DIR=/var/opt/repository-service-tuf/key_storage
+      - RSTUF_BROKER_SERVER=redis://redis
+      - RSTUF_REDIS_SERVER=redis://redis
+      - RSTUF_SQL_SERVER=mysql+pymysql://root:secret@mysql:3306/rstuf
```
Comment (MVrachev) [inline docker-compose-mysql.yml:64]: Use our new env names.
```suggestion
      - RSTUF_DB_SERVER=mysql+pymysql://root:secret@mysql:3306/rstuf
```

**Review by MVrachev — CHANGES_REQUESTED** [review]

File: docs/source/guide/Docker_README.md [hunk 3]
```diff
@@ -68,21 +68,26 @@ The result backend must to be compatible with Celery. See
 
 Example: `redis://redis`
 
-#### (Required) `RSTUF_SQL_SERVER`
+#### (Deprecated see `RSTUF_DB_SERVER`) `RSTUF_SQL_SERVER`
 
-RSTUF requires [PostgreSQL](https://www.postgresql.org).
+RSTUF requires [PostgreSQL](https://www.postgresql.org) or [MySQL/MariaDB](https://mariadb.org).
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/guide/Docker_README.md:73]: MariaDB is the open source of MySQL, it was a change from years ago.

File: docker-compose-mysql.yml (hunk 2 above)
Comment (kairoaraujo) [reply docker-compose-mysql.yml:64]: I added to keep it as a test for our deprecation test.
Let's keep it for now.

**Review by MVrachev — DISMISSED** [review]

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — DISMISSED** [review]

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — DISMISSED** [review]

**Comment by kairoaraujo** [comment]
Update to this PR:
Last night, while checking the Functional Tests, we identified an incompatibility with MySQL that requires more deep investigation.

The problem:
- RSTUF Worker receives a task to add/remove artifacts to TUF Metadata (main task)
- RSTUF Worker add/remove the artifact to DB
- RSTUF Worker creates the sub-task (children task) to publish the TUF Metadata in the storage and update the DB with Artifact published as True/1
- RSTUF sub-task finishes successfully, and the TUF Metadata is stored and DB Updated
- RSTUF Worker main task verifies if the requested artifacts are published in DB and remove from the pending publish list. It doesn't finish and the task stays in `RUNNING` state with `publishing` state.

The query in DB doesn't retrieve the updated artifact (`TargetFile`) object (it works for Postgres)
https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/12856541a5264204d6f5fc8bb899e8640bd571af/repository_service_tuf_worker/repository.py#L485-L487

**Comment by kairoaraujo** [comment]
> Update to this PR: Last night, while checking the Functional Tests, we identified an incompatibility with MySQL that requires more deep investigation.

This weekend I was able to fix it.
See 
https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/b0b01b5fae6c8d8b57c8ded0747807cd694b233d/repository_service_tuf_worker/repository.py#L511-L516

**Review by MVrachev — APPROVED** [review]

**Pull request merged**